x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/session/EqlSession.java

```diff
@@ -116,14 +120,25 @@ private <T> void preAnalyze(LogicalPlan parsed, ActionListener<LogicalPlan> list
             listener.onFailure(new TaskCancelledException("cancelled"));
             return;
         }
+        Set<String> fieldNames = fieldNames(parsed);
         indexResolver.resolveAsMergedMapping(
             indexWildcard,
+            fieldNames,
             configuration.indicesOptions(),
             configuration.runtimeMappings(),
             map(listener, r -> preAnalyzer.preAnalyze(parsed, r))
         );
     }
 
+    static Set<String> fieldNames(LogicalPlan parsed) {
+        Set<String> fieldNames = new HashSet<>();
+        parsed.forEachExpressionDown(
+            UnresolvedAttribute.class,
+            ua -> { fieldNames.add(ua.name().endsWith("*") ? ua.name() : ua.name() + "*"); }
```

Comment:
Should there be a `*` that is, we only care about the subfields? For field `foo` we want the subfields `foo.*` not everything that starts with foo (`foo*`).

Reply:
The PR in general tries to play it "safe" by improving the `field_caps` usage in the vast majority of cases. If we aim to be as exact as possible we might miss valid use cases that either we don't have tests for or we miss to create a valid test for it. I chose to look for the safer set of use cases to improve, rather than the very exact solution. With time, we can consider adding more PRs to improve the logic.

I didn't use `.*` because this might have further implications in EQL's "did you mean" user experience which might be considered a breaking change. Some further use cases and tests were needed for this scenario as well. In fact, we will lose "did you mean" functionality in some use cases. For example `equence with maxspan=10d [network where used == "ADMIN"] [network where id == 123]`. Assuming there is no `used` field but there is a `user` field - we will error out with a `verification_exception` and `Unknown column [used]` message but there won't be a `Unknown column [used], did you mean [user]?` anymore.

I will look into adding not `foo.*` but `foo` **and** `foo.*` (there is a slight but important difference between the two).